src/middlewares/methodWhitelist.ts

```diff
@@ -4,9 +4,25 @@ import { methods } from '../api'
 const allowedMethods = Object.keys(methods)
 
 export const methodWhitelist = (req: Request, res: Response, next: NextFunction) => {
-  const method = req.body?.method
-  if (method && allowedMethods.includes(method)) {
-    return next()
+  const body = req.body
+
+  if (Array.isArray(body)) {
```

Comment:
We are not validating for an empty body, which allows the code to enter the if statement (since an empty array is considered a valid array). Consequently, when the body array has no items, the allMethodsAllowed condition evaluates to true. This is because the Array.prototype.every() method returns true when invoked on an empty array, as there are no elements present that would violate the specified condition.

Reply:
If the body is an array, and has no items, there will be no method executed. 

Since there are no methods, all the methods are allowed.

Reply:
Zach added an empty array handler